File: Frontend/src/services/authService.js
import { apiRequest } from "../api/httpClient.js";
import { saveAuthData, clearAuthData } from "./authStorage.js";

/**
 * Đăng nhập người dùng bằng JWT.
 * Gọi API POST /api/v1/auth/login, nhận Access Token, Refresh Token, User Profile & Features Snapshot.
 */
export async function signIn({ username, password }) {
  const authResponse = await apiRequest("/api/v1/auth/login", {
    method: "POST",
    body: JSON.stringify({ username, password }),
  });

  saveAuthData({
    accessToken: authResponse.accessToken,
    refreshToken: authResponse.refreshToken,
    user: authResponse.user,
    features: authResponse.features,
  });

  return authResponse.user;
}

/**
 * Đăng xuất người dùng
 */
export function signOut() {
  clearAuthData();
}

export function getRoleHomePath(role) {
  switch (role) {
    case "SELLER":
      return "/seller/orders";
    case "SHIPPER":
      return "/shipper/orders";
    case "BUYER":
    default:
      return "/products";
  }
}

File: src/main/java/com/example/trackingorder/service/AuthService.java
package com.example.trackingorder.service;

import com.example.trackingorder.dto.request.LoginReq;
import com.example.trackingorder.dto.request.RefreshTokenReq;
import com.example.trackingorder.dto.response.AuthRes;

public interface AuthService {


//     Đăng nhập người dùng bằng username và password.
//     Trả về JWT Access Token (5 phút), Refresh Token, User Profile và Feature Snapshot map.
    AuthRes login(LoginReq req);


//     Làm mới Access Token bằng Refresh Token.
//     Trả về Access Token mới cùng Feature Snapshot mới nhất.
    AuthRes refresh(RefreshTokenReq req);

//  Lấy thông tin user hiện tại và Feature Snapshot tương ứng.
    AuthRes getMe();
}
